checkov/terraform/graph_builder/variable_rendering/evaluate_terraform.py

```diff
@@ -167,6 +172,96 @@ def evaluate_compare(input_str: str) -> Union[str, bool]:
     return input_str
 
 
+def _handle_literal(input_str: str) -> str:
+    try:
+        e = ast.literal_eval(input_str)
+        if isinstance(e, list) and len(e) == 1:
+            return e[0]
+    except (ValueError, SyntaxError):
+        return input_str
+
+
+def _handle_for_dict(rendered_foreach_statement: str, input_str: str, end_bracket_idx: int) -> Optional[str]:
+    try:
+        rendered_foreach_statement = json.loads(rendered_foreach_statement)
+    except JSONDecodeError:
+        return
+    return _handle_for_loop_in_dict(rendered_foreach_statement, input_str, end_bracket_idx + 1)
+
+
+def _handle_for_list(rendered_foreach_statement: str, input_str: str, end_bracket_idx: int) -> Optional[str]:
+    try:
+        rendered_foreach_statement = ast.literal_eval(rendered_foreach_statement.replace(' ', ''))
+    except (ValueError, SyntaxError):
+        return
+    return _handle_for_loop_in_list(rendered_foreach_statement, input_str, end_bracket_idx + 1)
```

Comment:
it looks like those methods just wrap `_handle_for_loop_in_dict` and `_handle_for_loop_in_list` methods and they are the only ones that call them, can you combine them?